In [10]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load in 

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the "../input/" directory.
# For example, running this (by clicking run or pressing Shift+Enter) will list the files in the input directory

import os
print(os.listdir("../input"))

# Any results you write to the current directory are saved as output.

['ExtractedTweets.csv', 'TwitterHandles.csv']


In [11]:
data = pd.read_csv(r"../input/ExtractedTweets.csv", encoding="latin1")
data = pd.concat([data.Party,data.Tweet],axis =1)
data.dropna(axis = 0, inplace = True)
data.Party = [1 if each == "Democrat" else 0 for each in data.Party]

In [12]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 86460 entries, 0 to 86459
Data columns (total 2 columns):
Party    86460 non-null int64
Tweet    86460 non-null object
dtypes: int64(1), object(1)
memory usage: 2.0+ MB


* Democrats    = 1 
* Respublicans = 0

* *I just checking our data.*

In [13]:
print(data)

       Party                                              Tweet
0          1  Today, Senate Dems vote to #SaveTheInternet. P...
1          1  RT @WinterHavenSun: Winter Haven resident / Al...
2          1  RT @NBCLatino: .@RepDarrenSoto noted that Hurr...
3          1  RT @NALCABPolicy: Meeting with @RepDarrenSoto ...
4          1  RT @Vegalteno: Hurricane season starts on June...
5          1  RT @EmgageActionFL: Thank you to all who came ...
6          1  Hurricane Maria left approx $90 billion in dam...
7          1  RT @Tharryry: I am delighted that @RepDarrenSo...
8          1  RT @HispanicCaucus: Trump's anti-immigrant pol...
9          1  RT @RepStephMurphy: Great joining @WeAreUnidos...
10         1  RT @AllSaints_FL: Zhihan (John)'s art received...
11         1  .@realDonaldTrump official policy to separate ...
12         1  Thank you to my mom Jean and all the mothers a...
13         1  We paid our respects at Nat’l Law Enforcemen...
14         1  RT @WinterHavenSun: Congra

In [14]:
import re

In [15]:
import nltk

In [16]:
from nltk.corpus import stopwords

In [17]:
import nltk as nlp

* For low words, i use ".lower()" metod and its changing our tweets words.

In [18]:
Tweet_list = []
for Tweet in data.Tweet:
    Tweet =re.sub("[^a-zA-Z]"," ",Tweet)
    Tweet = Tweet.lower()
    Tweet = nltk.word_tokenize(Tweet)
    Tweet = [ word for word in Tweet if not word in set(stopwords.words("english"))]
    lemma = nlp.WordNetLemmatizer()
    Tweet = [lemma.lemmatize(word) for word in Tweet]
    Tweet = " ".join(Tweet)
    Tweet_list.append(Tweet)

In [19]:
print(Tweet)

zika fear realized florida house gop acted prevent crisis dems inaction inexcusable time put politics aside amp work together


In [20]:
print(data.Tweet)

0        Today, Senate Dems vote to #SaveTheInternet. P...
1        RT @WinterHavenSun: Winter Haven resident / Al...
2        RT @NBCLatino: .@RepDarrenSoto noted that Hurr...
3        RT @NALCABPolicy: Meeting with @RepDarrenSoto ...
4        RT @Vegalteno: Hurricane season starts on June...
5        RT @EmgageActionFL: Thank you to all who came ...
6        Hurricane Maria left approx $90 billion in dam...
7        RT @Tharryry: I am delighted that @RepDarrenSo...
8        RT @HispanicCaucus: Trump's anti-immigrant pol...
9        RT @RepStephMurphy: Great joining @WeAreUnidos...
10       RT @AllSaints_FL: Zhihan (John)'s art received...
11       .@realDonaldTrump official policy to separate ...
12       Thank you to my mom Jean and all the mothers a...
13       We paid our respects at Nat’l Law Enforcemen...
14       RT @WinterHavenSun: Congratulations to  @AHSbl...
15       Meet 12 incredible young leaders appointed to ...
16       RT @wildlifeaction: With 1/3 of US wildlife in.

In [21]:
Tweet_list

['today senate dems vote savetheinternet proud support similar netneutrality legislation house http co n tggdlu l',
 'rt winterhavensun winter resident alta vista teacher one several recognized repdarrensoto national teacher apprecia',
 'rt nbclatino repdarrensoto noted hurricane maria left approximately billion damage congress allocated',
 'rt nalcabpolicy meeting repdarrensoto thanks taking time meet latinoleader ed marucci guzman nalcabpolicy',
 'rt vegalteno hurricane season start june st puerto rico readiness well pwr puertorico repdarrensoto espaillatny',
 'rt emgageactionfl thank came orlando gala successful night would possible without',
 'hurricane maria left approx billion damage yet billion allocated rebuilding grid surpr http co ku bckwuh',
 'rt tharryry delighted repdarrensoto voting cra overrule fcc save netneutrality rule find',
 'rt hispaniccaucus trump anti immigrant policy hurting small business across country find american willing',
 'rt repstephmurphy great joining 

In [39]:
from sklearn.feature_extraction.text import CountVectorizer
max_features = 5000

In [40]:
count_vectorizer = CountVectorizer(max_features=max_features , stop_words= "english")

In [41]:
sparce_matrix = count_vectorizer.fit_transform(Tweet_list).toarray()

In [42]:
print("Most used {} words: {}".format(max_features,count_vectorizer.get_feature_names()))

Most used 5000 words: ['aa', 'aapi', 'aarp', 'ab', 'abandon', 'abandoned', 'abc', 'abetterdeal', 'abiding', 'ability', 'able', 'abo', 'aboard', 'abortion', 'abou', 'abroad', 'absolute', 'absolutely', 'abuse', 'ac', 'aca', 'academic', 'academy', 'accept', 'acceptance', 'accepted', 'accepting', 'access', 'accessible', 'accident', 'accomplished', 'accomplishment', 'according', 'account', 'accountability', 'accountable', 'accurate', 'achieve', 'achievement', 'acknowledge', 'acre', 'act', 'acted', 'acti', 'acting', 'action', 'active', 'actively', 'activist', 'activity', 'actor', 'actually', 'ad', 'adam', 'add', 'added', 'addiction', 'adding', 'addition', 'additional', 'address', 'addressed', 'addressing', 'aderholt', 'adm', 'admin', 'administration', 'administrator', 'admission', 'admitted', 'adopted', 'adoption', 'adult', 'advance', 'advanced', 'advancing', 'advantage', 'advice', 'advisor', 'advisory', 'advocacy', 'advocate', 'advocated', 'advocating', 'ae', 'aerospace', 'af', 'afb', 'affa

## *Train Test Split*

In [43]:
y = data.iloc[:,0].values
x = sparce_matrix

In [44]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x,y, test_size = 0.1, random_state =42)

## *Naive Bayes*

In [45]:
from sklearn.naive_bayes import GaussianNB
nb = GaussianNB()
nb.fit(x_train,y_train)

GaussianNB(priors=None, var_smoothing=1e-09)

In [46]:
y_pred = nb.predict(x_test)
print("Accuracy of Tweets Democrats: ",nb.score(y_pred.reshape(-1,1),y_test))

Accuracy of Tweets Democrats:  0.7299329169558177


## Results Naive Bayes
> Our accuracy rate is 0.73, so if I use this key and i learn, in this data %73 Democrats generate.<
# *In data*
* %73 Democrats tweets 
* %27 Respublicans tweets